src/js/sonifier.js

```diff
@@ -0,0 +1,402 @@
+            }
+        },
+        // can override these to control synth behaviour
+        // These need better explanations of what they do
+        synthOptions: {
+            noteDurationConfig: {
+                divisorDuration: 3/8,
+                remainderDuration: 1/8
+            },
+            noteValueConfig: {
+                divisorValue: 91,
+                remainderValue: 89
+            },
+            envelopeDurationConfig: {
+                divisorDuration: 1/8,
+                divisorSilence: 1/4,
+                remainderDuration: 1/24,
+                remainderSilence: 1/12
+            },
+            envelopeValuesConfig: {
+                openValue: 1.0,
+                closedValue: 0.0
+            }
+        },
+        // options that control playback behaviour
+        playbackOptions: {
+            // Additional gap of silence between individual data points when
+            // playing a sequence
+            gapDuration: 1,
+            // Zoom factor - affects the durationConfigs of the synth when
+            // played. Higher numbers = slower playback
+            zoom: 1
+        },
+        invokers: {
+            "sonifyData": {
+                funcName: "floe.chartAuthoring.sonifier.dataEntriesToSonificationData",
+                args: "{that}"
+            },
+            "playSonification": {
+                funcName: "floe.chartAuthoring.sonifier.startSonification",
+                args: "{that}"
+            },
+            "stopSonification": {
+                funcName: "floe.chartAuthoring.sonifier.stopSonification",
+                args: "{that}"
+            },
+            "beginSonificationQueue": {
+                funcName: "floe.chartAuthoring.sonifier.processSonificationQueue",
+                args: [0, true, "{that}"]
+            }
+        },
+        events: {
+            // Fire at the end converting a data set to a sonification data
+            // set
+            onDataEntriesConvertedToSonificationData: null,
+            // Fires when a sonification play begins
+            onSonificationStarted: null,
+            // Fires when stopSonification function is called
+            onSonificationStopped: null
+        }
+    });
+
+    // Given an object in the style of floe.chartAuthoring.dataEntryPanel.model.dataEntries,
+    // convert it to an array of objects in the style used by the sonification components,
+    // maintaining object constancy by using the dataEntry object name as the key
+    floe.chartAuthoring.sonifier.dataEntriesToSonificationData = function(that) {
+        var unitDivisor = 10;
+        var sonificationData = floe.chartAuthoring.sonifier.unitDivisorSonificationStrategy(that, unitDivisor);
+        sonificationData.sort(floe.chartAuthoring.pieChart.legend.sortAscending);
+        that.applier.change("sonifiedData",sonificationData);
+        that.events.onDataEntriesConvertedToSonificationData.fire();
```

Comment:
Is there anyone who would listen to this event rather than simply listening to changes on "sonifiedData"?